experimental/Schemes/IdealSheaves.jl

```diff
@@ -335,3 +337,73 @@ function is_prime(I::IdealSheaf)
   return all(U->is_prime(I(U)), basic_patches(default_covering(space(I))))
 end
 
+function _minimal_power_such_that(I::Ideal, P::PropertyType) where {PropertyType}
+  whole_ring = ideal(base_ring(I), [one(base_ring(I))])
+  P(whole_ring) && return (0, whole_ring)
+  P(I) && return (1, I)
+  I_powers = [(1,I)]
+
+  while !P(last(I_powers)[2])
+    push!(I_powers, (last(I_powers)[1]*2, last(I_powers)[2]^2))
+  end
+  upper = pop!(I_powers)
+  lower = pop!(I_powers)
+  while upper[1]!=lower[1]+1
+    middle = pop!(I_powers)
+    middle = (lower[1]+middle[1], lower[2]*middle[2])
+    if P(middle[2])
+      upper = middle
+    else
+      lower = middle
+    end
+  end
+  return upper
+end
+
+@Markdown.doc """
+    order_on_divisor(f::VarietyFunctionFieldElem, I::IdealSheaf; check::Bool=true)
+
+Computes the order of a rational function `f` on a prime divisor 
+given by the ideal sheaf `I`.
+"""
+function order_on_divisor(
+    f::VarietyFunctionFieldElem, 
+    I::IdealSheaf;
+    check::Bool=true
+  )
+  if check
+    is_prime(I) || error("ideal sheaf must be a sheaf of prime ideals")
+  end
+  X = space(I)::AbsCoveredScheme
+  X == variety(parent(f)) || error("schemes not compatible")
+  
+  order_dict = Dict{AbsSpec, Int}()
+  for U in affine_charts(X)
+    # TODO: This shouldn't be necessary. Fix products of multiplicative sets.
```

Comment:
I would have expected that a single chart with `!(1 in I(U))` suffices for the computation.
Why do you take several? It should be the same order of vanishing in every chart.

Reply:
You probably know this better than me. It depends on whether or not we agree on the `IdealSheaf`s for the components to be prime or not. If we do (and we probably should, anyway), then maybe yes in the case of dimension > 1. I'm still a bit worried about curves, though.

Reply:
The order of vanishing needs a prime ideal. Otherwise it is not well defined. What worries you about curves?
So let us restrict to prime ideal sheaves for now.